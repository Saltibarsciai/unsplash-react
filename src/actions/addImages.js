const addImages = (newState) => {
    return {
        type: "addImages",
        newState
    }
};

export default addImages;