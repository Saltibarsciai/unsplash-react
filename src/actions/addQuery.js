const addQuery = (query) => {
    return {
        type: "addQuery",
        query
    }
};

export default addQuery;