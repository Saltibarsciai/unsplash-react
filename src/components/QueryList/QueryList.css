.query-list button {
  width: 100%;
  background: DarkSalmon;
  color: DarkRed;
  padding: 3px;
  margin-top: 5px;
  text-align: center;
  transition: 0.1s linear;
}
.query-list button:hover {
  background: #ffc4ad;
}
.query-list .slider {
  width: 100%;
  background: DarkRed;
  color: White;
  margin-top: -5px;
  text-align: center;
  transition: 0.1s linear;
  padding-top: 5px;
}
.query-list .slider > li {
  border-top: 1px solid DarkSalmon;
  list-style-type: none;
  cursor: pointer;
  margin: 0;
  padding: 5px;
}
.query-list .slider > li:hover {
  background-color: #d95224;
  color: White;
}

/*# sourceMappingURL=QueryList.css.map */
