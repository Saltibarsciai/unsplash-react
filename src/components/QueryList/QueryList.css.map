{"version":3,"sourceRoot":"","sources":["QueryList.scss","../../global-styles/_variables.scss"],"names":[],"mappings":"AAEE;EACE;EACA,YCJM;EDKN,OCHG;EDIH;EACA;EACA;EACA;;AACA;EACE,YCNQ;;ADSZ;EACE;EACA,YCdG;EDeH,OCdI;EDeJ;EACA;EACA;EACA;;AACA;EACE;EACA;EACA;EACA;EACA;;AACA;EACE,kBC5BI;ED6BJ,OC3BA","file":"QueryList.css"}