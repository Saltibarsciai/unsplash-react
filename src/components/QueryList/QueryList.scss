@import "../../global-styles/variables";
.query-list{
  button{
    width: 100%;
    background: $primary;
    color: $dark;
    padding: 3px;
    margin-top: 5px;
    text-align: center;
    transition: .1s linear;
    &:hover{
      background: $hover-light;
    }
  }
  .slider{
    width: 100%;
    background: $dark;
    color: $light;
    margin-top: -5px;
    text-align: center;
    transition: .1s linear;
    padding-top: 5px;
    >li{
      list-style-type: none;
      cursor: pointer;
      margin: 0;
      padding: 5px;
      &:hover{
        background-color: $secondary;
        color: $light;

      }
    }
  }
}