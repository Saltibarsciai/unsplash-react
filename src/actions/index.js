import addImages from './addImages'
import addQuery from './addQuery'

export {addImages, addQuery}