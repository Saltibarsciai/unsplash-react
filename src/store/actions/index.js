import addImages from './addImages'
import addQuery from './addQuery'
import {startLoader, stopLoader} from './loader'

export {
    addImages,
    addQuery,
    startLoader,
    stopLoader,
}
