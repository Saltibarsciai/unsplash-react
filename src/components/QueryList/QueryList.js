import React, {useState} from "react";
import Unsplash from "../../utils/Unsplash";
import {SlideDown} from "react-slidedown";
import {useSelector, useDispatch} from "react-redux";
import {replaceImages, startLoader, stopLoader} from "../../store/actions";
import "react-slidedown/lib/slidedown.css";
import "./QueryList.scss";

const QueryList = () => {
    const [isClosed, setClosed] = useState(true);
    const queryStore = useSelector(state => state.queryReducer);
    const dispatch = useDispatch();

    const fetchApi = (keywords) => {
        let id = "upload-images";
        dispatch(startLoader(id));
        Unsplash
            .fetch(keywords)
            .then( results => {
                dispatch(replaceImages(results));
                dispatch(stopLoader(id));
            });
    };

    const renderList = () => {
        return (
            queryStore.map((keywords, index) => (
                <li key={index} onClick={ () => fetchApi(keywords)}>
                    {keywords}
                </li>
            ))
        )
    };

    return (
        <section data-testid={"query-list"} className={"query-list"}>
            <button onClick={() => setClosed(!isClosed)}>
                Show saved queries
            </button>
            {/*test doesn't find prop - closed, had top put Closed also */}
            <SlideDown closed={isClosed} Closed={isClosed} className="slider" >
                {renderList()}
            </SlideDown>
        </section>
    );
};

export default QueryList;