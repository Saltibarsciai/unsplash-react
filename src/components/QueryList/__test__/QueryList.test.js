import React from 'react';
import renderer from 'react-test-renderer';
import QueryList from '../QueryList';
import configureStore from 'redux-mock-store';
import {Provider} from "react-redux";
import {startLoader} from "../../../store/actions";
import "@testing-library/jest-dom/extend-expect";
const mockStore = configureStore([]);
let component;

describe('QueryList component tests', () => {
    let store;
    beforeEach(() => {
        store = mockStore({
            queryReducer: [
                "Cars",
                "Moms",
                "Gym"
            ]
        });
        store.dispatch = jest.fn();
        component = renderer.create(
            <Provider store={store}>
                <QueryList />
            </Provider>
        );
    });

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should toggle slider', () => {
        const button = component.root.findByType('button');
        const slider = component.root.findByType('div');
        renderer.act(() => {
            button.props.onClick();
        });
        expect(slider.props.Closed).toEqual(false);

        renderer.act(() => {
            button.props.onClick();
        });
        expect(slider.props.Closed).toEqual(true);
    });

    it('should dispatch progress bar', () => {
        let button = component.root.findAllByType('li')[0];
        renderer.act(() => {
            button.props.onClick();
        });
        expect(store.dispatch).toHaveBeenCalledWith(
            startLoader("upload-images"));
    });
});