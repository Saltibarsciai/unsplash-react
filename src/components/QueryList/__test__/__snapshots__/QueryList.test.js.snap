// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<section
  className="query-list"
  data-testid="query-list"
>
  <button
    onClick={[Function]}
  >
    Show saved queries
  </button>
  <div
    Closed={true}
    className="react-slidedown slider"
    onTransitionEnd={[Function]}
  >
    <li
      onClick={[Function]}
    >
      Cars
    </li>
    <li
      onClick={[Function]}
    >
      Moms
    </li>
    <li
      onClick={[Function]}
    >
      Gym
    </li>
  </div>
</section>
`;

exports[`QueryList component tests should render with given state from Redux store 1`] = `
<section
  className="query-list"
  data-testid="query-list"
>
  <button
    onClick={[Function]}
  >
    Show saved queries
  </button>
  <div
    Closed={true}
    className="react-slidedown slider"
    onTransitionEnd={[Function]}
  >
    <li
      onClick={[Function]}
    >
      Cars
    </li>
    <li
      onClick={[Function]}
    >
      Moms
    </li>
    <li
      onClick={[Function]}
    >
      Gym
    </li>
  </div>
</section>
`;
